executor/point_get.go

```diff
@@ -354,16 +358,19 @@ func (e *PointGetExecutor) Next(ctx context.Context, req *chunk.Chunk) error {
 func (e *PointGetExecutor) getAndLock(ctx context.Context, key kv.Key) (val []byte, err error) {
 	if e.ctx.GetSessionVars().IsPessimisticReadConsistency() {
 		// Only Lock the exist keys in RC isolation.
-		val, err = e.get(ctx, key)
-		if err != nil {
-			if !kv.ErrNotExist.Equal(err) {
+		if e.lock {
```

Comment:
The original function doesn't check `e.lock`. What could be wrong if just do `lockKeyIfExists`?

Reply:
In the past, checking `e.lock` is in the function `lockKeyIfNeeded`. If e.lock is false, lockKeyIfNeeded does nothing and returns directly.
I keep the process as old that it gets the value in the function `getAndLock`.